src/pybind/mgr/dashboard/HACKING.rst

```diff
@@ -332,6 +332,87 @@ vs. ``it('image edit test' () => ...)``. As shown, the first example makes gramm
 prefix whereas the second message does not.``it()`` should describe what the individual test is doing and
 what it expects to happen.
 
+Differences between Frontend Unit Tests and End-to-End (E2E) Tests / FAQ
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+General introduction about testing and E2E/unit tests 
+
+
+What are E2E/unit tests designed for?
+.....................................
+
+E2E test:
+
+"Protractor is an end-to-end test framework for Angular and AngularJS applications.
+Protractor runs tests against your application running in a real browser,
+interacting with it as a user would." `(src) <http://www.protractortest.org/#/>`__
+
+It requires a fully functional system and tests the interation of all components
+of the application (Ceph, back-end, front-end).
+E2E tests are designed to mimic the behavior of the user when interacting with the application
+- for example when it comes to workflows like creating/editing/deleting an item.
+Also the tests should verify that certain items are displayed as a user would see them
+when clicking through the UI.
+
```

Comment:
This is not a comparison between two different tests, it's a document that's meant to explain when should be used which kind of test. I'd rather not add a comparison of both tests to this document.

Reply:
It's not about comparison it's about the usage and their benefits. There isn't a explanation why we need unit and functional tests and how it helps development. IMO as it is the HACKING.rst that is mend to be read by developers we should explain, how they and the code base benefit from the different types of test and what a pros and cons of both.

Reply:
Not sure what's the benefit of doing that as we need both. Of course you can add that yourself. Also note I may not agree with all points mentioned here. "very slow" is subjective and may also be biased. "full functional" doesn't hit the nail on the head, as some parts doesn't have to be enabled (and currently also can't be enabled on QA systems). Personally, I wouldn't describe racing issues, timing issues and other kinds of load sensitive issues as "random", as they occur due to certain conditions we should try to nail down. If a problem is hard to track down or not depends very much on how the tests have been written. I think we need to improve our code rather than add this as a downside of the tests.

Reply:
As per discussion I will provide information about the different tests scopes and usages for development in a separate issue afterwards, which will be discussed with a broader audience via a presentation.